### PR TITLE
Task/additional tests for validator gim 216

### PR DESCRIPTION
**Pull request by galinatremper**

### CODE REVIEW

File: tests/test_object_iterator.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    with open(file, encoding="utf8") as json_file:
+        data_json = yaml.safe_load(json_file)
+
+    # Example with the object embedded in the parent object
+    schema = BASE_DIR / "schemas" / "minimal_model_parent.yaml"
+    target_class = "ParentSubmission"
+
+    list1 = []
+    for elem in ObjectIterator(SchemaView(schema), data_json, target_class):
+        list1.append(elem)
```
Comment (lkuchenb) [inline tests/test_object_iterator.py]: Use list comprehension

File: tests/test_object_iterator.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+    # Example with the same object on the top level of JSON
+    data_json2 = data_json["submissions"][0]
+    schema = BASE_DIR / "schemas" / "minimal_model.yaml"
+    target_class = "Submission"
+
+    list2 = []
+    for elem in ObjectIterator(SchemaView(schema), data_json2, target_class):
+        list2.append(elem)
```
Comment (lkuchenb) [inline tests/test_object_iterator.py]: see above

**Review by lkuchenb — CHANGES_REQUESTED** [review]
Thanks, see comments inline. Apart form that, can we run the tests one level deeper without a report written to disk every time?

**Comment by galinatremper** [comment]
> Thanks, see comments inline. Apart form that, can we run the tests one level deeper without a report written to disk every time?

Well, we can certainly validate using one level deeper, but in this case we lose two-step validation, first json schema and then the rest. It would maybe make sense only for test cases, where we use only one plugin.

**Review by lkuchenb — APPROVED** [review]

**Pull request merged**